nipype/utils/filemanip.py

```diff
@@ -707,34 +707,33 @@ def loadpkl(infile):
     pkl_metadata = None
 
     unpkl = None
-    with indirectory(infile.parent):
-        with SoftFileLock('%s.lock' % infile.name):
```

Comment:
If we want to continue changing directories while correctly locking, what if you just switched (and locked the absolute path):

```diff
-    with indirectory(infile.parent):
-        with SoftFileLock('%s.lock' % infile.name):
+    with SoftFileLock('%s.lock' % infile):
+        with indirectory(infile.parent):

Reply:
This still has potential race conditions when it comes to loading the actual results file.